lib/models/builder.js

```diff
@@ -44,9 +44,14 @@ class Builder extends CoreObject {
    */
   readBuildFile(path) {
     // Load the build file
-    let buildFile = findBuildFile('ember-cli-build.js', path);
+    path = this.buildFilePath || path;
+
+    let project = this.project;
+    let projectRoot = project.root;
+
+    let buildFile = findBuildFile('ember-cli-build.js', { projectRoot, path });
     if (buildFile) {
-      return buildFile({ project: this.project });
+      return buildFile({ project });
     }
 
     throw new SilentError('No ember-cli-build.js found.');
```

Comment:
I also don't really think that the `findBuildFile` utility is very helpful at all.

I think we should do this (modulo any silly mistakes I've made while prototyping...):

```js
  readBuildFile() {
    // Load the build file
    let buildFile;
    if (this.buildFilePath) {
      buildFile = require(this.buildFilePath);
    } else {
      let buildFilePath = findUp.sync('ember-cli-build.js', { cwd: this.project.root });
      
      buildFile = require(buildFilePath);
    }

    if (typeof buildFile !== 'function') {
      throw new SilentError('Could not find a valid ember-cli-build.js file.');
    }

    process.chdir(projectRoot);

    return buildFile({ project });
  }
```

And delete that utility file.